packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -138,33 +143,35 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 	): void {
 		const postUpdateActions: (() => void)[] = [];
 		const audienceMembers = this.audience.getMembers();
+		const announcedAttendees = new Set<SessionClient>();
 		const connectedAttendees = new Set<SessionClient>();
 		for (const [clientConnectionId, value] of Object.entries(
 			remoteDatastore.clientToSessionId,
 		)) {
 			const clientSessionId = value.value;
-			const { attendee, isNew } = this.ensureAttendee(
+			const { attendee, isJoining } = this.ensureAttendee(
 				clientSessionId,
 				clientConnectionId,
 				/* order */ value.rev,
 			);
 
-			// Check new attendee against audience to see if they're currently connected
-			const isAttendeeConnected = audienceMembers.has(clientConnectionId);
+			const isAttendeeConnected =
+				// Attendee is connected if they are present in audience
+				audienceMembers.has(clientConnectionId) ||
+				// Attendee is connected if they are the sender of the update signal
+				senderConnectionId === clientConnectionId ||
+				// Attendee is connected if they were already marked as connected
+				connectedAttendees.has(attendee);
 
 			if (isAttendeeConnected) {
+				// If attendee is connected, update their connection ID and status.
 				connectedAttendees.add(attendee);
```

Comment:
Discussed changing here: https://github.com/microsoft/FluidFramework/pull/23113#discussion_r1844649212

Reply:
I'm going to remove/resolve this thread since the line range is off.
The question is meant for the comment and the call to setConnectionId. I don't see the answer to the question in linked comment.